peering/models/models.py

```diff
@@ -531,9 +532,21 @@ class DirectPeeringSession(BGPSession):
     router = models.ForeignKey(
         "Router", blank=True, null=True, on_delete=models.SET_NULL
     )
+    service_reference = models.CharField(
+        max_length=255,
+        unique=True,
+        blank=True,
+        null=True,
+        help_text="Optional: Internal Service Reference (will auto generate if left blank)",
+    )
```

Comment:
Set field in abstract class `BGPSession` so it is inherited in all subclasses.

No need to set null=True as field property, keeping it blank and as an empty string should be enough, unless it is a specific case to handle.